aws_lambda_powertools/utilities/feature_flags/appconfig.py

```diff
@@ -48,17 +55,29 @@ def __init__(
             Alternative JMESPath options to be included when filtering expr
         logger: A logging object
             Used to log messages. If None is supplied, one will be created.
+        boto_config: botocore.config.Config, optional
+            Botocore configuration to pass during client initialization
+        boto3_session : boto3.Session, optional
+            Boto3 session to use for AWS API communication
+        boto3_client : AppConfigDataClient, optional
+            Boto3 AppConfigDataClient Client to use, boto3_session and boto_config will be ignored if both are provided
```

Comment:
This naming looks better!
Initially I decided to stick to `sdk_client` because there was a `sdk_config` arg.
Shall we include a `logger.warning` discouraging the use of `sdk_config`?

Reply:
I thought about it, but I'd rather keep the behavior as is and deprecate it in Powertools v3.

Look at this issue I opened: https://github.com/aws-powertools/powertools-lambda-python/issues/4729

make sense?

Reply:
Looks perfect!